modules/compiler/utils/source/cl_builtin_info.cpp

```diff
@@ -1278,10 +1278,7 @@ Builtin CLBuiltinInfo::analyzeBuiltin(Function const &Callee) const {
     case eCLBuiltinWorkgroupScanLogicalXorExclusive:
       IsConvergent = true;
       Properties |= eBuiltinPropertyMapToMuxGroupBuiltin;
-      if (ID != eCLBuiltinWorkgroupAll && ID != eCLBuiltinWorkgroupAny &&
```

Comment:
why are we able to remove this check?

Reply:
Well my thinking was that __mux any/all _are_ overloaded so expect this to be set. But when I went back to look to address this comment, it turns out this function is just pushing stuff into a vector and never returning it, so it's all dead code. It might come from an earlier development version of the original PR.

The overload info that's important is being constructed afresh in `CLBuiltinInfo::mapGroupBuiltinToMuxGroupBuiltin`.

So I reckon we can just get rid of `OverloadInfo` completely from this function.